### PR TITLE
pkg/controller: Fix up node status.go docs

### PR DESCRIPTION
**Pull request by sgreene570**
**- What I did**
Cleaned up some inaccurate / hard to follow function doc strings.

**- How to verify it**

**- Description for the changelog**

Fix doc strings in pkg/controller node status

### CODE REVIEW

**Comment by runcom** [comment]
amazing
/approve
/lgtm

File: pkg/controller/node/status.go [hunk 1]
```diff
@@ -167,9 +167,8 @@ func isNodeMCDFailing(node *corev1.Node) bool {
 		isNodeMCDState(node, daemonconsts.MachineConfigDaemonStateUnreconcilable)
 }
 
-// getUpdatedMachines filters the provided nodes to ones whose current config
-// matches the desired config, which also matches the target config.
-// and the "done" flag is set.
```
Comment (runcom) [inline pkg/controller/node/status.go]: this is actully true if you look at `isNodeDoneAt`

File: pkg/controller/node/status.go [hunk 2]
```diff
@@ -221,8 +220,7 @@ func isNodeReady(node *corev1.Node) bool {
 	return checkNodeReady(node) == nil
 }
 
-// isNodeUnavailable is the backend for getUnavailableMachines;
-// see the docs for that for more information.
```
Comment (runcom) [inline pkg/controller/node/status.go]: any reason to drop this?

Comment (sgreene570) [reply pkg/controller/node/status.go]: I guess not. Can put back.

File: pkg/controller/node/status.go (hunk 1 above)
Comment (sgreene570) [reply pkg/controller/node/status.go]: I see. I thought having "desired config" and "target config" at the same time was confusing.

**Comment by kikisdeliveryservice** [comment]
Nice changes @sgreene570 !!!
